src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -1042,18 +1042,6 @@ impl LogicalAgg {
         having: Option<ExprImpl>,
         input: PlanRef,
     ) -> Result<(PlanRef, Vec<ExprImpl>, Option<ExprImpl>)> {
-        if select_exprs
-            .iter()
-            .chain(group_exprs.iter())
-            .any(|e| e.has_table_function())
-        {
-            return Err(ErrorCode::NotImplemented(
-                "Table functions in agg call or group by is not supported yet".to_string(),
-                3814.into(),
-            )
-            .into());
-        }
```

Comment:
This check was redundant and too restrictive:
* When table function is an input argument of agg call, or part of group by, it is already rejected by the `Project` node builder.
* When table function is a sibling of agg call, we are able to handle it.

Refer to planner tests in `agg.yaml` for examples of both cases above.